src/Sparrow.Server/Platform/RavenProcess.cs

```diff
@@ -0,0 +1,209 @@
+    {
+        public int ExitCode { get; set; }
+        public IntPtr Pid { get; set; }
+    }
+
+    public class RavenProcess : IDisposable
+    {
+        public ProcessStartInfo StartInfo { get; set; }
+        private bool _hasExited;
+
+        public event EventHandler ProcessExited;
+        private void OnProcessExited(EventArgs e)
+        {
+            EventHandler handler = ProcessExited;
+            handler?.Invoke(this, e);
+        }
+
+        public event Action<object, string> LineOutput;
+        private void OnLineOutput(string line, CancellationToken ctk)
+        {
+            if (ctk.IsCancellationRequested == false)
+            {
+                var handler = LineOutput;
+                handler?.Invoke(this, line);
+            }
+        }
+
+        public delegate void StreamWriteDelegate(Span<byte> bytes);
+
+        private readonly Logger _logger = LoggingSource.Instance.GetLogger<RavenProcess>("RavenProcess");
+        private IntPtr _pid = IntPtr.Zero;
+        private SafeFileHandle StandardOutAndErr { get; set; }
+        private SafeFileHandle StandardIn { get; set; }
+
+        public void Start(CancellationToken ctk)
+        {
+            if (StartInfo?.FileName == null)
+                throw new InvalidOperationException("RavenProcess Start() must be supplied with valid startInfo object and set Filename");
+
+            var rc = Pal.rvn_spawn_process(StartInfo.FileName, StartInfo.Arguments, out var pid, out var stdin, out var stdout, out var errorCode);
+            if (rc != PalFlags.FailCodes.Success)
+                PalHelper.ThrowLastError(rc, errorCode, $"Failed to spawn command '{StartInfo.FileName} {StartInfo.Arguments}'");
+
+            _pid = pid;
+            StandardOutAndErr = stdout;
+            StandardIn = stdin;
+        }
+
+        public static void Start(string filename, string arguments, StreamWriteDelegate outputDel)
+        {
+            using (var ravenProcess = new RavenProcess {StartInfo = new ProcessStartInfo {FileName = filename, Arguments = arguments}})
+            {
+                ravenProcess.Start(CancellationToken.None);
+                ravenProcess.ReadTo(outputDel);
+            }
+        }
+
+        private void ReadTo(StreamWriteDelegate outputDel)
+        {
+            var bytes = new byte[4096];
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            {
+                var read = fs.Read(bytes, 0, 4096);
+                while (read != 0)
+                {
+                    outputDel?.Invoke(new Span<byte>(bytes, 0, read));
+                    read = fs.Read(bytes, 0, 4096);
+                }
+            }
+        }
+
+        public Task<string> ReadToEndAsync()
+        {
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            using (var sr = new StreamReader(fs))
+            {
+                return sr.ReadToEndAsync();
```

Comment:
This is wrong here.
You are returning a task, but immediately disposing the stream before you actually read anything.